docs/proposals/202105256-cluster-class-and-managed-topologies.md

```diff
@@ -313,35 +557,125 @@ type LocalObjectTemplate struct {
       Replicas *int `json:"replicas,omitempty"`
     }
     ```
+1.  The `VariableTopology` object represents an instance of a variable.
+    ```golang
+    // VariableTopology can be used to customize the ClusterClass through
+    // patches. It must comply to the corresponding
+    // VariableDefinitionClass defined in the ClusterClass.
+    type VariableTopology struct {
+      // Name of the variable.
+      Name string `json:"name"`
+
+      // Value of the variable.
+      // Note: the value will be validated against the schema
+      // of the corresponding VariableDefinitionClass from the ClusterClass.
+      Value apiextensionsv1.JSON `json:"value"`
+    }
+    ```
+
+#### Builtin variables
+
+It’s also possible to use so-called builtin variables in addition to user-defined variables. The following builtin variables are available:
+- `cluster.{name,namespace,labels,annotations}`
+- `cluster.topology.{version,class}`
+- `cluster.topology.controlPlane.{labels,annotations,replicas}`
+    - **Note**: these variables are only available when patching control plane or control plane machine templates.
+- `cluster.topology.machineDeployment.current.{labels,annotations,replicas,class,name}`
```

Comment:
```suggestion
- `cluster.topology.machineDeployment.current.{labels,annotations,replicas,version,class}`
```
Adding version given that it could different that the topology one; removing name given that we apply patches to templates in MachineDeploymentClasses (and then the class should be the same for all the MachineDeployment instances we derive from it).